ddl/ddl_api.go

```diff
@@ -2218,23 +2245,32 @@ func (d *ddl) DropTablePartition(ctx sessionctx.Context, ident ast.Ident, spec *
 
 	err = d.doDDLJob(ctx, job)
 	if err != nil {
+		if ErrDropPartitionNonExistent.Equal(err) && spec.IfExists {
+			return nil
+		}
 		return errors.Trace(err)
 	}
 	err = d.callHookOnChanged(err)
 	return errors.Trace(err)
 }
 
 // DropColumn will drop a column from the table, now we don't support drop the column with index covered.
-func (d *ddl) DropColumn(ctx sessionctx.Context, ti ast.Ident, colName model.CIStr) error {
+func (d *ddl) DropColumn(ctx sessionctx.Context, ti ast.Ident, spec *ast.AlterTableSpec) error {
```

Comment:
Why don't we add an input parameter `IfExists bool` here?

Reply:
IMHO, use spec is better than add a new parameter.

Reply:
I prefer to use `spec`, because of someday more parameters in `spec` may be needed in the function.